### ISSUE TITLE
Duplication in advanced search

### ISSUE THREAD

**Issue by jospoortvliet** [nextcloud/server#43654]
Hi,

When you search, there are a lot of things that seem to duplicate things. What is the difference in searching in Collectives and it's pages etcetera? Maybe we can just stick to searching in Talk, Collectives, ...
<img width="277" alt="Screenshot 2024-02-14 at 13 42 51" src="https://github.com/nextcloud/server/assets/551757/c6a32a49-d6bd-4eec-a16e-2bb64dc49b2b">

I understand you might want to allow searching 'just' in the names of collectives, or just in the names of talk rooms, but that is already available in the sidebars of both apps, and it clutters this dialog a ton. Plus, it seems a real corner case to me.

**Comment by jancborchardt** [comment]
I agree, it should be separated by apps only, and we can then still in the content have sections for it. So e.g. you limit by "Collectives", and then in the content you have 3 sections:
- Collectives
- Pages
- Page content

With each having some results directly shown and the "Load more results" as last entry.

@sorbaugh @Fenn-CS @susnux is that doable?

**Comment by nimishavijay** [comment]
This also came up in the internal design feedback calls as an actual usability problem. People who are looking for something specific look for the app rather than the type of item itself. 

So if they are looking for a specific message, they would open this menu and scroll through it looking for "Talk" rather than "Messages". The lack of grouping and sorting in this is not super helpful to find what you are looking for either.

**Comment by Fenn-CS** [comment]
I would say it could be a usability issue but it's not a duplication from a technical standpoint. Does items do not show up all the time.

For example, "Messages in current Conversation" only show up if a conversation is open and I guess similar rules apply to rest but yeah, they could be some more smart grouping but the "feature" of letting people search for "Titles" for example conversation title, versus "content" should probably not be lost.

**Comment by nickvergessen** [comment]
With the upcoming filter options there is potential for a change?

**Comment by jancborchardt** [comment]
> the "feature" of letting people search for "Titles" for example conversation title, versus "content" should probably not be lost.

@Fenn-CS see my comment: https://github.com/nextcloud/server/issues/43654#issuecomment-1952503046
That wouldn’t make it lost but very visible still, while having only 1 entry in the dropdown per app.

**Comment by jospoortvliet** [comment]
> > the "feature" of letting people search for "Titles" for example conversation title, versus "content" should probably not be lost.
> 
> @Fenn-CS see my comment: [#43654 (comment)](https://github.com/nextcloud/server/issues/43654#issuecomment-1952503046) That wouldn’t make it lost but very visible still, while having only 1 entry in the dropdown per app.

exactly, it's about the drop-down. I'm not proposing to remove the searching in these different places, just to unify the option. I can hardly think of a use case where one would want to search in titles of Talk rooms only vs also in comments. Yes, when you get too many results - I get it, it can happen. But 95% of the time it only complicates things and clutters the UI.
